JRC Publications Repository.js

```diff
@@ -0,0 +1,477 @@
+			return "journalArticle";
+		}
+	}
+	if ( getSearchResults(doc, true) ) {
+		return "multiple";
+	}
+}
+
+//We need a different name for the same function, because of calling another translator inside scrape makes problems.
+function detectWebHere(doc, url) {
+	var type = ZU.xpath(doc, '//meta[contains(@name, "DC.type")]/@content');
+	if (type.length>0) {
+		if (mappingTable[type[0].textContent]) {
+			return mappingTable[type[0].textContent];
+		} else {//generic fallback
+			Z.debug('Unrecognized item type: ' + type[0].textContent);
+			return "journalArticle";
+		}
+	}
+	if ( getSearchResults(doc, true) ) {
+		return "multiple";
+	}
+}
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return true;
+			}
+			var articles = new Array();
+			for (var i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
+}
+
+function scrape(doc, url) {
+	//Calling detectWeb here on multiples, does only work for the first one?!
+	Z.debug( 'DETECTWEB = ' + detectWeb(doc, url) );//I guess this then tries (for the nth (n>1) entry) to use the detectWeb function form the called Metadata translator.
+	Z.debug( 'DETECTWEBHERE = ' + detectWebHere(doc, url) );//This always uses the correct function.
+	var itemType = detectWebHere(doc, url);
+	
+	var trans = Zotero.loadTranslator('web');
+	trans.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');//https://github.com/zotero/translators/blob/master/Embedded%20Metadata.js
+	trans.setDocument(doc);
+	
+	trans.setHandler('itemDone', function (obj, item) {
+		
+		item.itemType = itemType;
+		
+		if (item.title == item.title.toUpperCase()) {
+			item.title = ZU.capitalizeTitle( item.title, true );
+		}
+		
+		//The format of the authors in the meta-tags is very special (pattern = LASTNAME Firstname):
+		//e.g. <meta name="DC.creator" content="DEL RIO GONZALEZ Pablo" xml:lang="en_GB">
+		//We try to overwrite the creators with better splitting of lastName and firstName.
+		var authors = ZU.xpath(doc, '//meta[@name="DC.creator" and @content]|//meta[@name="DC.contributor" and @content]');
+		if (authors) {
+			item.creators = [];
+			for (var a=0; a<authors.length; a++) {
+				var authorsText = authors[a].content;
+				var authorsType = authors[a].name.substr(3);//either creator or contributer
+				var lastName = "";
+				var firstName = "";
+				var authorParts = authorsText.split(' ');
+				for (var b=0; b<authorParts.length; b++) {
+					//distinguish between lastName (every letter is in uppercase) from firstName
+					//but there might also be just initials (e.g. "D.") from the firstName
+					if (authorParts[b].toUpperCase() == authorParts[b] && authorParts[b].indexOf('.') == -1) {
+						lastName += ' ' + ZU.capitalizeTitle(authorParts[b], true)
+					} else {
+						firstName += ' ' + authorParts[b];
+					}
+				}
```

Comment:
Yeah, you don't have to use regexp and I did go with the assumption (possibly wrong) that the names were always in the form LAST First. The logic above should be clearer, because I don't think your intention (although this may not occur in practice) is to turn "SMITH John F" into "Smith F, John" (i.e. initials don't always get periods and the last name/first name blocks should be contiguous)